policies/templates/gcp_app_service_versions.yaml

```diff
@@ -0,0 +1,70 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+apiVersion: templates.gatekeeper.sh/v1alpha1
+kind: ConstraintTemplate
+metadata:
+  name: gcp-app-service-versions-v1
+spec:
+  crd:
+    spec:
+      names:
+        kind: GCPAppEngineServiceVersionsConstraintV1
+        plural: gcpappengineserviceversionsconstraintsv1
+      validation:
+        openAPIV3Schema:
+          properties: {}
+  targets:
+   validation.gcp.forsetisecurity.org:
+      rego: | #INLINE("validator/app_service_versions.rego")
+           #
+           # Copyright 2018 Google LLC
+           #
+           # Licensed under the Apache License, Version 2.0 (the "License");
+           # you may not use this file except in compliance with the License.
+           # You may obtain a copy of the License at
+           #
+           #      http://www.apache.org/licenses/LICENSE-2.0
+           #
+           # Unless required by applicable law or agreed to in writing, software
+           # distributed under the License is distributed on an "AS IS" BASIS,
+           # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+           # See the License for the specific language governing permissions and
+           # limitations under the License.
+           #
+           
+           package templates.gcp.GCPAppEngineServiceVersionsConstraintV1
+           
+           import data.validator.gcp.lib as lib
+           
+           deny[{
+           	"msg": message,
+           	"details": metadata,
+           }] {
+           	constraint := input.constraint
+           	params := lib.get_constraint_params(constraint)
+           
+           	# Restrict to App Engine services
+           	asset := input.asset
+           	asset.asset_type == "appengine.googleapis.com/Service"
+           
+           	# Check version count
+           	max_versions := lib.get_default(params, "max_versions", 2)
```

Comment:
Why 2 by default? I am not sure what should be a good default here

Reply:
2 supports having a "next" or canary version and the current production version.

Reply:
sounds good